e2e/claims.service.e2e.ts

```diff
@@ -809,4 +870,118 @@ describe('Сlaim tests', () => {
       ).toBeDefined();
     });
   });
+
+  describe('Verification Tests', () => {
+    const createExampleSignedCredential = async (
+      issuerFields: IssuerFields[],
+      namespace: string,
+      expirationDate?: Date
+    ) => {
+      return await verifiableCredentialsService.createRoleVC({
+        id: rootOwnerDID,
+        namespace: namespace,
+        version: '1',
+        issuerFields,
+        expirationDate,
+      });
+    };
+
+    test('verifyVc should verify a credential with no errors if the Issuer is authorized', async () => {
+      await signerService.connect(rootOwner, ProviderType.PrivateKey);
+      await domainsService.createRole({
+        roleName: verifyVcRole,
+        namespace,
+        data: roles[`${verifyVcRole}.${root}`],
+        returnSteps: false,
+      });
+      const vc = await createExampleSignedCredential(
+        [],
+        `${verifyVcRole}.${root}`
+      );
+      const result = await claimsService.verifyVc(vc);
+      expect(result.errors).toHaveLength(0);
+      expect(result.isVerified).toBeTruthy;
+    });
+    test('verifyVc should: not verify a credential and return an issuer error if the issuer is not authorized', async () => {
+      const roleName = `${verifyVcRole2}.${root}`;
+      await domainsService.createRole({
+        roleName: verifyVcRole2,
+        namespace,
+        data: roles[roleName],
+        returnSteps: false,
+      });
+      const vc = await createExampleSignedCredential([], roleName);
+      const result = await claimsService.verifyVc(vc);
+      expect(result.errors).toHaveLength(1);
+      expect(result.errors).toContain(
+        `Issuer ${rootOwnerDID} is not authorized to issue ${roleName}: issuer is not in DID list`
+      );
+      expect(result.isVerified).toBeFalsy;
+    });
+    test('resolveCredentialAndVerify should resolve a Verifiable Credential and call correct verification method (verifyVC)', async () => {
+      const roleName = `${resolveVC}.${root}`;
+      console.log(rootOwnerDID, 'CHECK THE DID');
+      await signerService.connect(rootOwner, ProviderType.PrivateKey);
+      await domainsService.createRole({
+        roleName: resolveVC,
+        namespace,
+        data: roles[roleName],
+        returnSteps: false,
+      });
+      await createExampleSignedCredential([], roleName);
+      const result = await claimsService.resolveCredentialAndVerify(
+        rootOwnerDID,
+        roleName
+      );
+      console.log(result, 'THE RESOLVE RESULT');
+    });
+    test('verifyOffChainClaim should verify an issued off chain claim', async () => {
+      const roleName = `${verifyOffChainClaimRole}.${root}`;
```

Comment:
@JGiter THis is the correct way to create an 'offChainClaim' I think. The claim is created (see below), but the message always has the issuer as 0x000 so I set it as the issuer as the Issuer DID before appending it. The issuer is always verified successfully (claimIssuerVerifier.verifyIssuer) but the claimIssuerVerifier.verifyIssuance always fails:

```
    const token = await this._credentialResolver.getClaimIssuedToken(
      subject,
      role
    );
    if (!token) {
      throw new Error(
        'Unable to resolve the issuer credential to verify their authority'
      );
    }
```

Reply:
The below is the claim I get issued. Does not look like this has the same interface of OffChainClaim I can pass to verifyOffChainClaim


```
    {
      id: '07857751-1dc8-4c35-94f8-ba7b69651b99',
      requester: 'did:ethr:volta:0x2BC1ad4BcDc54Ea17f5355448F0200019DB34797',
      claimIssuer: [ 'did:ethr:volta:0xFE5300613484147de721607cBc753B356ecCA2Be' ],
      acceptedBy: 'did:ethr:volta:0xFE5300613484147de721607cBc753B356ecCA2Be',
      onChainProof: '0x0cea25f65ef4bcb2a560c0e3046741f9433153ee2d6346318f8d8c4eeec1ff6e44050197dfe1fc78d5a8d5fce283b403fdd058a1022139ee6074692a58f197681b',
      issuedToken: 'eyJhbGciOiJFUzI1NksiLCJ0eXAiOiJKV1QifQ.eyJjbGFpbURhdGEiOnsiY2xhaW1UeXBlIjoidmVyaWZ5T25DaGFpbi5ld2MiLCJjbGFpbVR5cGVWZXJzaW9uIjoxLCJyZXF1ZXN0b3JGaWVsZHMiOlt7ImtleSI6InRlbXBlcmF0dXJlIiwidmFsdWUiOjM2fV0sImlzc3VlckZpZWxkcyI6W119LCJkaWQiOiJkaWQ6ZXRocjp2b2x0YToweDJCQzFhZDRCY0RjNTRFYTE3ZjUzNTU0NDhGMDIwMDAxOURCMzQ3OTciLCJzaWduZXIiOiJkaWQ6ZXRocjp2b2x0YToweEZFNTMwMDYxMzQ4NDE0N2RlNzIxNjA3Y0JjNzUzQjM1NmVjQ0EyQmUiLCJpc3MiOiJkaWQ6ZXRocjp2b2x0YToweEZFNTMwMDYxMzQ4NDE0N2RlNzIxNjA3Y0JjNzUzQjM1NmVjQ0EyQmUiLCJzdWIiOiJkaWQ6ZXRocjp2b2x0YToweDJCQzFhZDRCY0RjNTRFYTE3ZjUzNTU0NDhGMDIwMDAxOURCMzQ3OTcifQ.MHhkOTMyMWZkMzNiOWRkM2QyNWY0ZDk1NGNlYTJhNDViZmZhZDBlYjFlN2RlMDU0M2VkZGUxYWE3YWZkMWY5ZDJjMGZlMjM3YzIyOGE3NWJkMzg4YzFlMWQxNmNlM2RjMWJlZjYxMjJmODNlNGI4Mzc0YTBkM2MzY2U2MmQ0YzkyMzFj',
      vp: '{"@context":["https://www.w3.org/2018/credentials/v1"],"id":"urn:uuid:e578ca5b-7473-480a-a3be-30227c8ffda4","type":["VerifiablePresentation"],"holder":"did:ethr:0x12047:0xfe5300613484147de721607cbc753b356ecca2be","verifiableCredential":[{"@context":["https://www.w3.org/2018/credentials/v1"],"id":"urn:uuid:8244ebe4-6661-4a7e-bd31-85fc358c7109","type":["VerifiableCredential","EWFRole"],"credentialSubject":{"id":"did:ethr:volta:0x2BC1ad4BcDc54Ea17f5355448F0200019DB34797","role":{"namespace":"verifyOnChain.ewc","version":"1"},"issuerFields":[]},"issuer":"did:ethr:0x12047:0xfe5300613484147de721607cbc753b356ecca2be","issuanceDate":"2022-07-11T11:07:58.675Z","proof":{"@context":"https://w3id.org/security/suites/eip712sig-2021/v1","type":"EthereumEip712Signature2021","proofPurpose":"assertionMethod","proofValue":"0x8efff22d795b84786492eeb7ae414009746890d94fa8def9e10b0095485ee09133acf6a593385c48327185804ee8593124e2a4f6da01f8fcdceeaacfaf11cf3e1b","verificationMethod":"did:ethr:0x12047:0xfe5300613484147de721607cbc753b356ecca2be#controller","created":"2022-07-11T11:07:58.676Z","eip712Domain":{"domain":{},"messageSchema":{"CredentialSubject":[{"name":"id","type":"string"},{"name":"role","type":"EWFRole"},{"name":"issuerFields","type":"IssuerFields[]"}],"EIP712Domain":[],"EWFRole":[{"name":"namespace","type":"string"},{"name":"version","type":"string"}],"IssuerFields":[{"name":"key","type":"string"},{"name":"value","type":"string"}],"Proof":[{"name":"@context","type":"string"},{"name":"verificationMethod","type":"string"},{"name":"created","type":"string"},{"name":"proofPurpose","type":"string"},{"name":"type","type":"string"}],"StatusList2021Entry":[{"name":"id","type":"string"},{"name":"type","type":"string"},{"name":"statusPurpose","type":"string"},{"name":"statusListIndex","type":"string"},{"name":"statusListCredential","type":"string"}],"VerifiableCredential":[{"name":"@context","type":"string[]"},{"name":"id","type":"string"},{"name":"type","type":"string[]"},{"name":"issuer","type":"string"},{"name":"issuanceDate","type":"string"},{"name":"credentialSubject","type":"CredentialSubject"},{"name":"proof","type":"Proof"},{"name":"credentialStatus","type":"StatusList2021Entry"}]},"primaryType":"VerifiableCredential"}},"credentialStatus":{"id":"https://energyweb.org/credential/0x57bea3ed273c6b72caa78b5ef9a0884811c24a6151b431df28ab8796f8c1f6bd#list","type":"StatusList2021Entry","statusPurpose":"revocation","statusListIndex":"1","statusListCredential":"https://identitycache.org/v1/status-list/urn:uuid:8244ebe4-6661-4a7e-bd31-85fc358c7109"}}],"proof":{"@context":"https://w3id.org/security/suites/eip712sig-2021/v1","type":"EthereumEip712Signature2021","created":"2022-07-11T11:07:58.734Z","verificationMethod":"did:ethr:0x12047:0xfe5300613484147de721607cbc753b356ecca2be#controller","proofPurpose":"authentication","proofValue":"0xc3c2b78992a078fc50f43bf3ac56aabf7e32007b4ca59fbb793e3f1b3835247f55985f83b200757c7c2b1332000b7cd0b292409e0ad25efe2d11ee7d2e45a5db1c","eip712Domain":{"domain":{},"messageSchema":{"VC712DomainTypedDataField":[{"name":"name","type":"string"},{"name":"type","type":"string"}],"VC712DomainData":[],"VC712DomainSchema":[{"name":"StatusList2021Entry","type":"VC712DomainTypedDataField[]"},{"name":"CredentialSubject","type":"VC712DomainTypedDataField[]"},{"name":"EIP712Domain","type":"VC712DomainTypedDataField[]"},{"name":"EWFRole","type":"VC712DomainTypedDataField[]"},{"name":"IssuerFields","type":"VC712DomainTypedDataField[]"},{"name":"Proof","type":"VC712DomainTypedDataField[]"},{"name":"VerifiableCredential","type":"VC712DomainTypedDataField[]"}],"VC712Domain":[{"name":"domain","type":"VC712DomainData"},{"name":"messageSchema","type":"VC712DomainSchema"},{"name":"primaryType","type":"string"}],"EWFRole":[{"name":"namespace","type":"string"},{"name":"version","type":"string"}],"IssuerFields":[{"name":"key","type":"string"},{"name":"value","type":"string"}],"CredentialSubject":[{"name":"id","type":"string"},{"name":"role","type":"EWFRole"},{"name":"issuerFields","type":"IssuerFields[]"}],"StatusList2021Entry":[{"name":"id","type":"string"},{"name":"type","type":"string"},{"name":"statusPurpose","type":"string"},{"name":"statusListIndex","type":"string"},{"name":"statusListCredential","type":"string"}],"VCProof":[{"name":"@context","type":"string"},{"name":"verificationMethod","type":"string"},{"name":"created","type":"string"},{"name":"proofPurpose","type":"string"},{"name":"type","type":"string"},{"type":"string","name":"proofValue"},{"type":"VC712Domain","name":"eip712Domain"}],"VerifiableCredential":[{"type":"string[]","name":"@context"},{"type":"string","name":"id"},{"type":"string[]","name":"type"},{"type":"string","name":"issuer"},{"type":"string","name":"issuanceDate"},{"type":"CredentialSubject","name":"credentialSubject"},{"type":"StatusList2021Entry","name":"credentialStatus"},{"type":"VCProof","name":"proof"}],"Proof":[{"name":"@context","type":"string"},{"name":"verificationMethod","type":"string"},{"name":"created","type":"string"},{"name":"proofPurpose","type":"string"},{"name":"type","type":"string"}],"VerifiablePresentation":[{"type":"string[]","name":"@context"},{"type":"string","name":"id"},{"type":"string[]","name":"type"},{"type":"string","name":"holder"},{"type":"VerifiableCredential[]","name":"verifiableCredential"},{"type":"Proof","name":"proof"}],"EIP712Domain":[]},"primaryType":"VerifiablePresentation"}}}'
    } 
```

Reply:
@whitneypurdum Sorry, I think I made a mistake. I think that the signature of `verifyOffChainClaim` should be `async verifyOffChainClaim(eip191JWT: string): Promise<CredentialVerificationResult>`

And then you can use the `jsonwebtoken` package to decode the body. https://github.com/energywebfoundation/iam-client-lib/blob/757de5a4458b0864d621cbb41c44f764c87b3b5c/src/modules/claims/claims.service.ts#L3
where a decoded body should look something like: 
```
{
  "claimData": {
    "claimType": "myrole.roles.myorg.iam.ewc",
    "claimTypeVersion": 1
  },
  "signer": "did:ethr:volta:0xb764D50751fF83e66FCF5A08Fb0De3f830248728",
  "iss": "did:ethr:volta:0xb764D50751fF83e66FCF5A08Fb0De3f830248728",
  "sub": "did:ethr:volta:0x0fAe481148C2DF77886CE3874f0fb2E7118acDa0"
}
```

Reply:
but @jrhender just so we are clear, the getCredential method will return the OffChainClaim interface. Is it the issuedToken value I should be then passing to verifyOffChainClaim:
```
export interface OffChainClaim {
    claimType: string;
    claimTypeVersion: number;
    issuedToken: string;
    iss: string;
    [x: string]: string | number;
}
```

Reply:
I am using the enrol and issue method to generate a claim. A claim is generated (see returned JSON below), so these are the keys I have available to pass on to verifyOffChainClaim. However, once I successfully enrol and issue the claim, and then call getCredential, I never get a resolved credential back. 


```
    {
      id: 'd55aaa5a-aed2-4679-9848-13d350563a20',
      requester: 'did:ethr:volta:0x9E574FAf6AeCeCe397414D18b7988f82cC30E3B7',
      claimIssuer: [ 'did:ethr:volta:0x1Ed12D7A770DEC1a40d52d22d4B9428CF0a5a170' ],
      acceptedBy: 'did:ethr:volta:0x1Ed12D7A770DEC1a40d52d22d4B9428CF0a5a170',
      onChainProof: '0xcb0b1d45518acd887eb5bff8348abcbc37f865f2e249383bdd84f4ac172144da1ae4d1eb99abc98c0775232b654d48fbfb964e8474b77195df923f2b0af7f94a1c',
      issuedToken: 'eyJhbGciOiJFUzI1NksiLCJ0eXAiOiJKV1QifQ.eyJjbGFpbURhdGEiOnsiY2xhaW1UeXBlIjoidmVyaWZ5T25DaGFpbi5ld2MiLCJjbGFpbVR5cGVWZXJzaW9uIjoxLCJyZXF1ZXN0b3JGaWVsZHMiOlt7ImtleSI6InRlbXBlcmF0dXJlIiwidmFsdWUiOjM2fV0sImlzc3VlckZpZWxkcyI6W119LCJkaWQiOiJkaWQ6ZXRocjp2b2x0YToweDlFNTc0RkFmNkFlQ2VDZTM5NzQxNEQxOGI3OTg4ZjgyY0MzMEUzQjciLCJzaWduZXIiOiJkaWQ6ZXRocjp2b2x0YToweDFFZDEyRDdBNzcwREVDMWE0MGQ1MmQyMmQ0Qjk0MjhDRjBhNWExNzAiLCJpc3MiOiJkaWQ6ZXRocjp2b2x0YToweDFFZDEyRDdBNzcwREVDMWE0MGQ1MmQyMmQ0Qjk0MjhDRjBhNWExNzAiLCJzdWIiOiJkaWQ6ZXRocjp2b2x0YToweDlFNTc0RkFmNkFlQ2VDZTM5NzQxNEQxOGI3OTg4ZjgyY0MzMEUzQjcifQ.MHhiOTMzOGRlMWQyNzg3OWQyMGIxYjM2OGIyYzYwNTI4NDRiNDAwOTIzODYxODA2ZmIwODc4ODNmNTllNGFmZDg4NjM0NTM4YTYyZDhmZjAwMTQ0YjNjMGFhNGViMjg5MjY2MDVmNjM0NjVhYjY0YjAxNGMxZWZkNTMwZWY5MzdkOTFi',
      vp: '{"@context":["https://www.w3.org/2018/credentials/v1"],"id":"urn:uuid:eb03d18b-a868-4f00-b801-54f192392f0d","type":["VerifiablePresentation"],"holder":"did:ethr:0x12047:0x1ed12d7a770dec1a40d52d22d4b9428cf0a5a170","verifiableCredential":[{"@context":["https://www.w3.org/2018/credentials/v1"],"id":"urn:uuid:a55478d0-4e94-4a5b-ae0d-4b39d9f0374d","type":["VerifiableCredential","EWFRole"],"credentialSubject":{"id":"did:ethr:volta:0x9E574FAf6AeCeCe397414D18b7988f82cC30E3B7","issuerFields":[],"role":{"namespace":"verifyOnChain.ewc","version":"1"}},"issuer":"did:ethr:0x12047:0x1ed12d7a770dec1a40d52d22d4b9428cf0a5a170","issuanceDate":"2022-07-11T13:18:25.539Z","proof":{"@context":"https://w3id.org/security/suites/eip712sig-2021/v1","type":"EthereumEip712Signature2021","proofPurpose":"assertionMethod","proofValue":"0x2362e1482bebd77671914abee6d94995af159649ebd187092cc85731a4a8d02c65cb5505ea25bcca6bb00f0252416947ac30ce146d0fffa7c42df9b86d317d901b","verificationMethod":"did:ethr:0x12047:0x1ed12d7a770dec1a40d52d22d4b9428cf0a5a170#controller","created":"2022-07-11T13:18:25.540Z","eip712Domain":{"domain":{},"messageSchema":{"CredentialSubject":[{"name":"id","type":"string"},{"name":"role","type":"EWFRole"},{"name":"issuerFields","type":"IssuerFields[]"}],"EIP712Domain":[],"EWFRole":[{"name":"namespace","type":"string"},{"name":"version","type":"string"}],"IssuerFields":[{"name":"key","type":"string"},{"name":"value","type":"string"}],"Proof":[{"name":"@context","type":"string"},{"name":"verificationMethod","type":"string"},{"name":"created","type":"string"},{"name":"proofPurpose","type":"string"},{"name":"type","type":"string"}],"StatusList2021Entry":[{"name":"id","type":"string"},{"name":"type","type":"string"},{"name":"statusPurpose","type":"string"},{"name":"statusListIndex","type":"string"},{"name":"statusListCredential","type":"string"}],"VerifiableCredential":[{"name":"@context","type":"string[]"},{"name":"id","type":"string"},{"name":"type","type":"string[]"},{"name":"issuer","type":"string"},{"name":"issuanceDate","type":"string"},{"name":"credentialSubject","type":"CredentialSubject"},{"name":"proof","type":"Proof"},{"name":"credentialStatus","type":"StatusList2021Entry"}]},"primaryType":"VerifiableCredential"}},"credentialStatus":{"id":"https://energyweb.org/credential/0x2005ee33e59cd233c74b249649c80cd8f21c07cd84ecca4c2ac29f47d692ea13#list","type":"StatusList2021Entry","statusPurpose":"revocation","statusListIndex":"1","statusListCredential":"https://identitycache.org/v1/status-list/urn:uuid:a55478d0-4e94-4a5b-ae0d-4b39d9f0374d"}}],"proof":{"@context":"https://w3id.org/security/suites/eip712sig-2021/v1","type":"EthereumEip712Signature2021","created":"2022-07-11T13:18:25.624Z","verificationMethod":"did:ethr:0x12047:0x1ed12d7a770dec1a40d52d22d4b9428cf0a5a170#controller","proofPurpose":"authentication","proofValue":"0xf0745b761a11339ca718d052bf56129cd259dd5a70dd4c99f74b799d94a86b7f7073921b141532e87b3ea1e3c58b352a955d7e41412cbcf85ea7ee427b3a8b281b","eip712Domain":{"domain":{},"messageSchema":{"VC712DomainTypedDataField":[{"name":"name","type":"string"},{"name":"type","type":"string"}],"VC712DomainData":[],"VC712DomainSchema":[{"name":"StatusList2021Entry","type":"VC712DomainTypedDataField[]"},{"name":"CredentialSubject","type":"VC712DomainTypedDataField[]"},{"name":"EIP712Domain","type":"VC712DomainTypedDataField[]"},{"name":"EWFRole","type":"VC712DomainTypedDataField[]"},{"name":"IssuerFields","type":"VC712DomainTypedDataField[]"},{"name":"Proof","type":"VC712DomainTypedDataField[]"},{"name":"VerifiableCredential","type":"VC712DomainTypedDataField[]"}],"VC712Domain":[{"name":"domain","type":"VC712DomainData"},{"name":"messageSchema","type":"VC712DomainSchema"},{"name":"primaryType","type":"string"}],"EWFRole":[{"name":"namespace","type":"string"},{"name":"version","type":"string"}],"IssuerFields":[{"name":"key","type":"string"},{"name":"value","type":"string"}],"CredentialSubject":[{"name":"id","type":"string"},{"name":"role","type":"EWFRole"},{"name":"issuerFields","type":"IssuerFields[]"}],"StatusList2021Entry":[{"name":"id","type":"string"},{"name":"type","type":"string"},{"name":"statusPurpose","type":"string"},{"name":"statusListIndex","type":"string"},{"name":"statusListCredential","type":"string"}],"VCProof":[{"name":"@context","type":"string"},{"name":"verificationMethod","type":"string"},{"name":"created","type":"string"},{"name":"proofPurpose","type":"string"},{"name":"type","type":"string"},{"type":"string","name":"proofValue"},{"type":"VC712Domain","name":"eip712Domain"}],"VerifiableCredential":[{"type":"string[]","name":"@context"},{"type":"string","name":"id"},{"type":"string[]","name":"type"},{"type":"string","name":"issuer"},{"type":"string","name":"issuanceDate"},{"type":"CredentialSubject","name":"credentialSubject"},{"type":"StatusList2021Entry","name":"credentialStatus"},{"type":"VCProof","name":"proof"}],"Proof":[{"name":"@context","type":"string"},{"name":"verificationMethod","type":"string"},{"name":"created","type":"string"},{"name":"proofPurpose","type":"string"},{"name":"type","type":"string"}],"VerifiablePresentation":[{"type":"string[]","name":"@context"},{"type":"string","name":"id"},{"type":"string[]","name":"type"},{"type":"string","name":"holder"},{"type":"VerifiableCredential[]","name":"verifiableCredential"},{"type":"Proof","name":"proof"}],"EIP712Domain":[]},"primaryType":"VerifiablePresentation"}}}'
    }
```